src/components/Header/Header.jsx

```diff
@@ -12,7 +16,7 @@ const Header = ({ title }) => {
       .then((res) => res.json())
       .then((data) => {
         // console.log(data.main.temp.toFixed());
-        setData(data);
+        setTemp(data);
```

Comment:
You're setting `temp` in 2 places (also on line 9). 
It makes sense to only save the data when you do the API call. 

In `changeCity` you could be saving the _city_ name, which you will _use to make_ the API call.